### PR TITLE
BUG GH11693 Support NaT series concatenation

### LINKED ISSUE #11693: Inconsistent concat behavior between datetime64[ns] and tz-aware version in 0.17.1
This edge case appears when concatenating a timezone aware datetime series with another that is filled with only pd.NaT.  This works if the second series is only partially filled with pd.NaT.  It also works if the second series is another timezone (in that case the resulting Series is timezone unaware).

```python
import pandas as pd
pd.__version__
x = pd.Series( pd.date_range('20151124 08:00', '20151124 09:00', freq = '1h') )
y = pd.Series( pd.date_range('20151124 10:00', '20151124 11:00', freq = '1h') )
pd.concat([x,y]) # Works

y[:] = pd.NaT
pd.concat([x,y]) # Works

x = pd.Series( pd.date_range('20151124 08:00', '20151124 09:00', freq = '1h', tz = "US/Eastern") )
y = pd.Series( pd.date_range('20151124 10:00', '20151124 11:00', freq = '1h', tz = "US/Eastern") )
pd.concat([x,y])

y[0] = pd.NaT
pd.concat([x,y]) # Works

y[:] = pd.NaT
pd.concat([x,y]) # Fails
```

### PR DESCRIPTION
**Pull request by varunentropy**
Please review .

Closes #11693

### CODE REVIEW

File: pandas/tools/tests/test_merge.py [hunk 1]
```diff
@@ -2255,6 +2255,16 @@ def test_with_mixed_tuples(self):
...
         result = concat([df1,df2])
 
+    def test_concat_Nat_series(self):
+        # GH 11693
+        # test for merging NaT series with datetime series.
+        x = pd.Series( pd.date_range('20151124 08:00', '20151124 09:00', freq = '1h', tz = "US/Eastern") )
+        y = pd.Series( pd.date_range('20151124 10:00', '20151124 11:00', freq = '1h', tz = "US/Eastern") )
+        y[:] = pd.NaT
+        xy = pd.concat([x,y])
+        tm.assert_equal(xy[0], x[0])
```
Comment (jreback) [inline pandas/tools/tests/test_merge.py]: use ``tm.assert_series_equal``

``self.assertEqual(xy[2], pd.NaT)``

File: pandas/tslib.pyx [hunk 2]
```diff
@@ -3423,6 +3423,9 @@ def tz_convert(ndarray[int64_t] vals, object tz1, object tz2):
     trans, deltas, typ = _get_dst_info(tz2)
     trans_len = len(trans)
 
+    if (utc_dates==iNaT).all():
+        return utc_dates
+
```
Comment (jreback) [inline pandas/tslib.pyx]: add this as an or condition to the lines below as its checking the same thing

e.g.

```
if (result==iNAT).all() or (utc_dates==iNaT).all():
    ....
```

File: pandas/tslib.pyx [hunk 3]
```diff
@@ -3423,6 +3423,9 @@ def tz_convert(ndarray[int64_t] vals, object tz1, object tz2):
     trans, deltas, typ = _get_dst_info(tz2)
     trans_len = len(trans)
```
Comment (jreback) [inline pandas/tslib.pyx]: add a comment explaining the condtions this is checking

**Comment by jreback** [comment]
couple comments, pls add a whatsnew note

File: pandas/tools/tests/test_merge.py [hunk 4]
```diff
@@ -2255,6 +2255,16 @@ def test_with_mixed_tuples(self):
         df2 = DataFrame({ u'B' : 'foo', (u'B',1) : 'bar' },index=range(2))
         result = concat([df1,df2])
 
+    def test_concat_Nat_series(self):
+        # GH 11693
+        # test for merging NaT series with datetime series.
+        x = pd.Series( pd.date_range('20151124 08:00', '20151124 09:00', freq = '1h', tz = "US/Eastern") )
+        y = pd.Series( pd.date_range('20151124 10:00', '20151124 11:00', freq = '1h', tz = "US/Eastern") )
+        y[:] = pd.NaT
+        xy = pd.concat([x,y])
```
Comment (jreback) [inline pandas/tools/tests/test_merge.py]: not 100% sure that we have all of these tests, so can you add this for no-tz's as well, and both series all-nat (both with and w/o tz's). use a loop to check all 4 conditions.

Comment (jreback) [inline pandas/tools/tests/test_merge.py]: further a concat of tz-aware and (different tz or no tz) should result in ``object`` (works, just put all tests in the same place).

Comment (jreback) [inline pandas/tools/tests/test_merge.py]: if we have other tests in merge which test this kind of thing, pls move to a similar location (I don't know if we do)

**Comment by jreback** [comment]
can you update according to comments

**Comment by varun-kr** [comment]
apologies for the delay. Will fix over the weekend.

File: pandas/tools/merge.py [hunk 5]
```diff
@@ -955,8 +955,12 @@ def get_result(self):
 
             # stack blocks
             if self.axis == 0:
-                new_data = com._concat_compat([x._values for x in self.objs])
+                to_concat = [x._values for x in self.objs]
+                typs = com.get_dtype_kinds(to_concat)
```
Comment (jreback) [inline pandas/tools/merge.py]: this should be fixed solely in ``_concat_compat``

**Comment by jreback** [comment]
can you update

**Comment by jreback** [comment]
superseded by #12195

**Pull request closed without merging**